src/TimeModeling/judiVector.jl

```diff
@@ -709,6 +709,29 @@ iterate(S::judiVector, state::Integer=1) = state > S.nsrc ? nothing : (S.data[st
 
 ####################################################################################################
 
+# Integration/differentiation of shot records
+
+function cumsum(x::judiVector;dims=1)
+    dims == 1 || dims == 2 || throw(judiVectorException("dims is out of range, please double check your dims argument"))
```

Comment:
More explicit error would be nice like
" Dimension number $(dims) out of range for a two-dimensional array"

But in theory, like 718 will throw an error anyway if dims is not 1 or 2 so this may not be necessary,

Reply:
Sure! Modified this according to your suggestion in latest commit